skfp/model_selection/splitters/maxmin_split.py

```diff
@@ -0,0 +1,291 @@
+import warnings
+from collections.abc import Sequence
+from numbers import Integral
+from typing import Any, Optional, Union
+
+from rdkit.Chem import Mol
+from rdkit.Chem.rdFingerprintGenerator import GetMorganGenerator
+from rdkit.SimDivFilters.rdSimDivPickers import MaxMinPicker
+from sklearn.utils._param_validation import Interval, RealNotInt, validate_params
+
+from skfp.model_selection.splitters.utils import (
+    ensure_nonempty_subset,
+    get_data_from_indices,
+    split_additional_data,
+    validate_train_test_split_sizes,
+    validate_train_valid_test_split_sizes,
+)
+from skfp.utils.validators import ensure_mols
+
+
+@validate_params(
+    {
+        "data": ["array-like"],
+        "additional_data": ["tuple"],
+        "train_size": [
+            Interval(RealNotInt, 0, 1, closed="neither"),
+            Interval(Integral, 1, None, closed="left"),
+            None,
+        ],
+        "test_size": [
+            Interval(RealNotInt, 0, 1, closed="neither"),
+            Interval(Integral, 1, None, closed="left"),
+            None,
+        ],
+        "return_indices": ["boolean"],
+    },
+    prefer_skip_nested_validation=True,
+)
+def maxmin_train_test_split(
+    data: Sequence[Union[str, Mol]],
+    *additional_data: Sequence,
+    train_size: Optional[float] = None,
+    test_size: Optional[float] = None,
+    return_indices: bool = False,
+    random_state: int = 0,
+) -> Union[
+    tuple[
+        Sequence[Union[str, Mol]], Sequence[Union[str, Mol]], Sequence[Sequence[Any]]
+    ],
+    tuple[Sequence, ...],
+    tuple[Sequence[int], Sequence[int]],
+]:
+    """
+    Split using MaxMin algorithm.
+
+    MaxMinPicker is an efficient algorithm for picking a optimal subset of diverse compounds from a candidate pool.
+    The algorithm is described in Ashton, M. [1]_,implementation was created by Roger Sayle [3]_ [4]_ [5]_
+    at the 2017 RDKit user group meeting [2]_.[4]_
```

Comment:
1. We generally don't mention papers explicitly. You can reword this as `The original algorithm was introduced in [1]_, but here we use an optimized implementation by Roger Sayle [2]_ [3]_ [4]_ [5]_`
2. Don't mention RDKit UGM, it's already in reference.
3. Don't put any references after the dot
4. There is no reference 5?